config.c

```diff
@@ -154,8 +154,7 @@ char *config_path_for_type(enum config_type type, const char *name)
 	ret = stat(config, &sbuf);
 	if ((ret == -1 && errno == ENOENT) || !S_ISDIR(sbuf.st_mode)) {
 		unlink(config);
-		if (mkdir(config, 0700) < 0)
```

Comment:
Correct me if I'm wrong but lpass creates logs, lock files, (that it later uses for blob sync with the server)etc, in that folder. So if it fails to create to folder ( or can not write to it ) it is possible the user will not have the blob data synced to the lastpass servers. ( For example make_upload_dir function in upload-queue.c )